### PR TITLE
updated test_full config 

### PR DESCRIPTION
**Pull request by cying111**
# nf-core/nanoseq pull request

Many thanks for contributing to nf-core/nanoseq!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/nanoseq branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/nanoseq)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/nanoseq/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -14,7 +14,6 @@ def helpMessage() {
     log.info"""
     Usage:
     The typical command for running the pipeline is as follows:
-
```
Comment (drpatelh) [inline main.nf:17]: All of these line keep on being deleted for some reason 😅 Can we add them back in please.

File: main.nf [hunk 2]
```diff
@@ -860,6 +849,8 @@ if (!params.skip_alignment) {
     ch_sortbam_quant
         .map { it -> [ it[0], it[3] ] }
         .set { ch_sortbam_quant }
+    ch_sample_map_deseq2 = "F"
+    ch_sample_map_dexseq = "F"
```
Comment (drpatelh) [inline main.nf:853]:
```suggestion
```

Comment (drpatelh) [reply main.nf:853]: What are these extra variables doing? The information in the processes I wrote should be enough to do a differential analysis. Just extract the group info directly from the counts matrix within the R script.

Comment (drpatelh) [reply main.nf:853]: You should just need to change the R scripts and uncomment out the code here to test. No need to change any of the logic.

File: bin/run_dexseq.r [hunk 3]
```diff
@@ -139,4 +148,4 @@ suppressWarnings({
   dex.padj <- getAdjustedPValues(stageRObj, order=FALSE,
                                  onlySignificantGenes=FALSE)
 })
-write.csv(cbind(dxr_pval,dex.padj$gene,dex.padj$transcript), file="dexseq.results.txt")
+write.csv(cbind(dxr_pval,dex.padj$gene,dex.padj$transcript), file="DEXseq_out.txt")
```
Comment (drpatelh) [inline bin/run_dexseq.r:151]:
```suggestion
write.csv(cbind(dxr_pval,dex.padj$gene,dex.padj$transcript), file="dexseq.results.txt")
```

Comment (drpatelh) [reply bin/run_dexseq.r:151]: Better to have it named like this. I have also documented it like this in `output.md`

**Review by drpatelh — CHANGES_REQUESTED** [review]
Some changes to the differential analysis code guys.

File: main.nf (hunk 2 above)
Comment (yuukiiwa) [reply main.nf:853]: Hi @drpatelh, with the adjusted Rscript without the samplesheet parameter, the DESeq2 and the DEXSeq processes ran fine with fastq inputs but not bam inputs. For the fastq inputs, the sorted.bam file names inputted into bambu and stringtie2+featurecounts are renamed to Group_Replicate (ex. MCF7_R1), but for the bam inputs, the bam file names are not renamed, and bambu and stringtie2+featureCounts are using the full name of the bam files. The column names of count outputs from bambu/stringtie2+featureCounts from fastq and bam inputs are different: fastq inputs have column names in the format of Group_Replicate (ex.MCF7_R1), whereas bam inputs are using the full names of the input files, which can be anything, as column names.

File: bin/run_dexseq.r (hunk 3 above)
Comment (yuukiiwa) [reply bin/run_dexseq.r]: Hi @drpatelh, we have changed it to "dexseq.results.txt".

File: main.nf (hunk 1 above)
Comment (yuukiiwa) [reply main.nf]: Hi @drpatelh, we have added those empty lines back in.

File: main.nf [hunk 4]
```diff
@@ -869,10 +871,13 @@ if (!params.skip_alignment) {
     ch_samplesheet_reformat
         .splitCsv(header:true, sep:',')
         .map { get_sample_info(it, params.genomes) }
-        .map { it -> if (it[1].toString().endsWith('.bam')) [ it[0], it[1] ] }
-        .set { ch_sortbam_quant }
+        .map { it -> if (it[1].toString().endsWith('.bam')) [ it[0] , it[1] ] }
+        .into { ch_sortbam_quant
+                ch_sample_map_deseq2
+                ch_sample_map_dexseq }
```
Comment (drpatelh) [inline main.nf:877]:
```suggestion
        .into { 
            ch_sortbam_rename
            ch_sortbam_quant
        }
 
 Hmmm...yes. Good point 🤔 Maybe the most consistent thing to do is to add an additional process to rename the BAM files [here](https://github.com/nf-core/nanoseq/blob/85f2ef340183096f34b9a30e692be978b4769cb6/main.nf#L873)? and then pass those files to the downstream processes?

Something like this:

```
process BAM_RENAME {
    tag "$sample"

    input:
    tuple val(sample), path(bam) from ch_sortbam_rename

    output:
    tuple val(sample), path("*.bam") into ch_sortbam_quant

    script:
    """
    [ ! -f ${sample}.bam ] && ln -s $bam ${sample}.bam
    """
}
```
```

Comment (drpatelh) [reply main.nf:877]: Hmmm...yes. Good point 🤔 Maybe the most consistent thing to do is to add an additional process to rename the BAM file like I have done above by adding a small process, and then pass those files to the downstream processes?

Comment (yuukiiwa) [reply main.nf]: Hi @drpatelh, thank you for the modification! We have added the process you wrote to the pull request and modified the Rscripts for DESeq2 and DEXSeq.

**Review by drpatelh — APPROVED** [review]
Amazing! Thanks guys 🙂

**Comment by yuukiiwa** [comment]
Thank you so much, @drpatelh!!

**Pull request merged**